### PR TITLE
fix codacy issues, part 2

### PR DESCRIPTION
**Pull request by pazos**
Remove static context, which is unsafe beecause it can't be garbage collected
Pass application context unless interaction with the screen is needed.
Log catched Java exceptions as errors. 

~~it seems to work, but I need a few days to test.~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ ant debug
 
 ## To-do
 
-* a concept to deal with native Lua modules.
-* a loader for native modules that have been put into the activity's library directory?
-* a loader for obb storage, maybe? We could put native Lua modules there, for example.
-* example code for framebuffer access and more
+-   a concept to deal with native Lua modules.
```
Comment (Frenzie) [inline README.md]: Please don't change this. That's ugly af and seriously decreases readability. :-/

(It's *their* default.)

Comment (pazos) [reply README.md]: yeah, it looks ugly. I don't know if we can suppress certain warnings with a configuration file.

Comment (Frenzie) [reply README.md]: If not we can just ignore the README file wholesale if desired. I don't find it very pressing either way. :-)

Comment (pazos) [reply README.md]: I love green A's and hate those orange E letters. :cry:

**Pull request merged**